includes/class-wcpdf-main.php

```diff
@@ -594,44 +594,25 @@ public function load_custom_and_local_fonts() {
 		require WPO_WCPDF()->plugin_path() . '/vendor/autoload.php';
 
 		$options = array(
-			'defaultFont'				=> 'dejavu sans',
-			'tempDir'					=> $this->get_tmp_path( 'dompdf' ),
-			'logOutputFile'				=> $this->get_tmp_path( 'dompdf' ) . "/log.htm",
-			'fontDir'					=> $this->get_custom_fonts_path(),
-			'fontCache'					=> $this->get_custom_fonts_path(),
-			'isRemoteEnabled'			=> true,
-			'isFontSubsettingEnabled'	=> true,
-			'isHtml5ParserEnabled'		=> true,
+			'defaultFont'             => 'dejavu sans',
+			'tempDir'                 => $this->get_tmp_path( 'dompdf' ),
+			'logOutputFile'           => $this->get_tmp_path( 'dompdf' ) . "/log.htm",
+			'fontDir'                 => $this->get_tmp_path( 'fonts' ),
+			'fontCache'               => $this->get_tmp_path( 'fonts' ),
+			'isRemoteEnabled'         => true,
+			'isFontSubsettingEnabled' => true,
+			'isHtml5ParserEnabled'    => true,
 		);
 
-		if ( ! @is_dir( $this->get_dompdf_fonts_path() ) || ! wp_is_writable( $this->get_dompdf_fonts_path() ) ) {
-			$rootDir = $this->get_custom_fonts_path();
-		}
-
-		// save custom fonts
-		$custom_fonts_options = new Options( $options );
-		if( isset( $rootDir ) ) {
-			$custom_fonts_options->setRootDir( $rootDir );
-		}
-		$dompdf = new Dompdf( $custom_fonts_options );
-		$dompdf->getFontMetrics()->saveFontFamilies();
-
-		// save local fonts
-		$options['fontDir']  = $options['fontCache'] = $this->get_tmp_path( 'fonts' );
-		$local_fonts_options = new Options( $options );
-		if( isset( $rootDir ) ) {
-			$local_fonts_options->setRootDir( $rootDir );
-		}
-		$dompdf->getFontMetrics()->setOptions( $local_fonts_options );
+		$rootDir     = $this->get_custom_fonts_path();
+		$options_obj = new Options( $options );
+		$options_obj->setRootDir( $rootDir );
+		$dompdf      = new Dompdf( $options_obj );
 		$dompdf->getFontMetrics()->saveFontFamilies();
 	}
 
 	public function get_custom_fonts_path() {
-		return trailingslashit( WPO_WCPDF()->plugin_path() ) . 'assets' . DIRECTORY_SEPARATOR . 'fonts';
-	}
-
-	public function get_dompdf_fonts_path() {
-		return trailingslashit( WPO_WCPDF()->plugin_path() ) . 'vendor' . DIRECTORY_SEPARATOR . 'dompdf' . DIRECTORY_SEPARATOR . 'dompdf' . DIRECTORY_SEPARATOR . 'lib' . DIRECTORY_SEPARATOR . 'fonts' . DIRECTORY_SEPARATOR;
+		return trailingslashit( WPO_WCPDF()->plugin_path() ) . 'assets/fonts';
```

Comment:
dompdf will be looking in a specific folder relative to this path if you use this as root path, `$rootDir . "/lib/fonts/dompdf_font_family_cache.dist.php"` ([source](https://github.com/dompdf/dompdf/blob/8768448244967a46d6e67b891d30878e0e15d25c/src/FontMetrics.php#L132))

So I doubt that this works with the current code?

Reply:
I followed the same folder structure, see the `assets/fonts` directory

Reply:
I see now - I only saw the font files themselves which are not in that path. I think it's cleaner to keep everything together like the original

Reply:
But then we have an issue here: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/c9e3489f548e186f5fd12002731908eec3a04bff/vendor/dompdf/dompdf/src/FontMetrics.php#L132

When passing the `rootDir` here: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/c9e3489f548e186f5fd12002731908eec3a04bff/includes/class-wcpdf-main.php#L609

Reply:
What I meant is to keep all the fonts in the `assets/fonts/lib/fonts/` dir as well, together with the `dompdf_font_family_cache.dist.php` file (and adapting the `$distfonts` variable accordingly)